proposals/0003.md

```diff
@@ -0,0 +1,122 @@
+`radicle-dev` must comply with the following sections.
+
+# Recommendation
+
+In this section we include the templates that need to be included and also other
+recommendations to make the licensing experience as smooth as possible for both
+contributors and core developers.
+
+## License File
+
+Every repository MUST include a `LICENSE` file which corresponds to this
+[file][license-file]. As well as this, there MUST be a `COPYING` file which is a
+symlink to the `LICENSE` file.
+
+## DCO File
+
+Every repository MUST include a `DCO` which corresponds to this
+[file][dco-file].
+
+## License Header
+
+Every file that contains source code MUST include a header which corresponds to
+this [file][license-header] at the **top** of each source file as a comment.
+
+**Note** that there are placeholders `<project name>` which correspond to the
+name of the repository. You MUST replace these with the actual name.
+
+Also included is a [script][license-header-check] for checking the
+license-header for Rust files.
+
+## CONTRIBUTING File
+
+Every repository MUST include a `CONTRIBUTING.md` file. The
+inclusion of the `CONTRIBUTING.md` file will mean that GitHub will prompt new
+contributors to look at the file before making a pull request.
+
+The file must include the following section:
+
+```
+## Certificate of Origin
+
+By contributing to this project, you agree to the [Developer Certificate of
+Origin (DCO)](./path/to/DCO). This document was created by the Linux Kernel community and is a
+simple statement that you, as a contributor, have the legal right to make the
+contribution.
+
+In order to show your agreement with the DCO you should include at the end of the commit message,
+the following line: Signed-off-by: John Doe <john.doe@example.com>, using your real name and email.
+
+This can be done easily using the -s flag on the git commit.
+
+If you find yourself pushed a few commits without Signed-off-by, you can still add it afterwards.
+
+### Fixing the DCO
+
+If you did not sign one or more of your commits then it is not all for not. The
+crowd at `src-d` have a wonderful guide on how to remedy this situation
+[here](https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository)
+
+## License Header
+
+As part of our license, we must include a license header at the top of each
+source file. The template for this header can be found
+[here](./path/to/license-header). If you are creating a new file in the
+repository you will have to add this header to the file.
+```
+
+## DCO Bot
+
+To ensure that all commits by an outside contributor are signed we SHALL add a
+[DOC bot][dco-bot] to monitor pull requests on our repositories.
+
+## Merging Pull Requests
+
+We SHALL NOT merge pull requests using the GitHub UI. This is due to GitHub
+creating commits on our behalf using a bot. These commits will not be signed.
```

Comment:
I’m wondering whether we want to specify the actual process for this in this document. For example whether we use merge commits are fast-forwards and who pushes the changes to master. Alternatively we can do this later in the different repos.

Reply:
I think this section is a bit inaccurate: technically, a merge commit which does not alter the `tree` does not bear any copyrightable contribution _per definitionem_, and thus does not need a sign-off. It is, however, not entirely clear if THE GREEN BUTTON always does the right thing wrt to `Signed-off-by`, especially when the branch to be merged contains commits by multiple authors. Therefore, not using THE GREEN BUTTON is preferable.

I do **not** think we actually have to require that all commits be _cryptographically_ signed, only release tags (but that is a separate discussion).

Not wanting commits made by GitHub (which, coincidentally, _are_ cryptographically signed) is an entirely separate issue, and only a matter of policy.